src/store/root.ts

```diff
@@ -0,0 +1,68 @@
+import { enableMapSet } from 'immer';
+import { CustomMarket } from 'src/ui-config/marketsConfig';
+import create from 'zustand';
+import { devtools } from 'zustand/middleware';
+
+import { createGovernanceSlice, GovernanceSlice } from './governanceSlice';
+import { createIncentiveSlice, IncentiveSlice } from './incentiveSlice';
+import { createPoolSlice, PoolSlice } from './poolSlice';
+import { createProtocolDataSlice, ProtocolDataSlice } from './protocolDataSlice';
+import { createStakeSlice, StakeSlice } from './stakeSlice';
+import { createSingletonSubscriber } from './utils/createSingletonSubscriber';
+import { getQueryParameter } from './utils/queryParams';
+import { createWalletSlice, WalletSlice } from './walletSlice';
+
+enableMapSet();
+
+export type RootStore = StakeSlice &
+  ProtocolDataSlice &
+  WalletSlice &
+  PoolSlice &
+  IncentiveSlice &
+  GovernanceSlice;
+
+export const useRootStore = create<RootStore>()(
+  devtools((...args) => {
+    return {
+      ...createStakeSlice(...args),
+      ...createProtocolDataSlice(...args),
+      ...createWalletSlice(...args),
+      ...createPoolSlice(...args),
+      ...createIncentiveSlice(...args),
+      ...createGovernanceSlice(...args),
+    };
```

Comment:
My feedback on this large PR is one thing - these slices should belong to a top-level key.  Similar to how a large global redux store scoped actions and reducers within a given slice to a top-level key. There are a couple reasons that this is useful:

1. When creating new methods and/or state values within any given slice, a developer now needs to know if there is a key within that slice exists in some other slice across some other file within the codebase, which creates opportunity for human-error and can create more time debugging.  Now if/when I want to add a new value/method to a given slice, I now have to check across other slice files to ensure that that name doesn't exist already, which is cognitive overload.  It could also cause side-effects - i.e. what happens when there are two keys with the same name within the global store, does one get overwritten?  This is problematic, and scoping the slices to a top-level key prevents this.  Common keys that I see this are ones like `address`, `account`, `balance`, etc.
2. When using `useRootStore` at the component level, now it becomes very clear as to where that value/method belongs to.  It can help with debugging, it can help to understand that X method is part of Y feature/part of the protocol.

I'd like to see something like this implemented:

```js
// alphabetized (ocd)
return {
  governance: ...createGovernanceSlice(...args),
  incentive: ...createIncentiveSlice(...args),
  market: ...createMarketSlice(...args), // see previous comments
  network: ...createNetworkSlice(...args), // see previous comments
  pool: ...createPoolSlice(...args),
  staking: ...createStakeSlice(...args),
  wallet: ...createWalletSlice(...args),
};
 ```
 
 This will require the changes across each `useRootStore` use, but this is a good change.   It becomes clear on what part of the store you're working with.
 
 ```js
 // this reads nicely
 const governance = useRootStore(state => state.governance)
 governance.submitVote({...})
 
 // or, syntax sugar
 const { submitVote } = useRootStore(state => state.governance)
 submitVote({...})
 ```

Reply:
Awesome feedback, I think it's a good idea, but just for a sake of different perspective, if you have 2 methods with the same name, but different parameters type check will not pass. 
Splitting code to different reducers was the hardest part of Redux IMO, and I love that in zustand it's way simpler to access different parts of store, so you don't need to think too hard in terms of context of each reducer/slice but rather always have a global store at your disposal at any given moment. 
It's still way easier to maintain than using context where everything is coupled to hooks and local state, but at the same time very simple abstraction layer without too many specifics. But I think having keyed slices is worth it anyway to avoid name collision of methods with the same signature

Reply:
Personally i think this is a rather bad idea and is counter to the core ideas of zustand (being stupid simple, easy to debug global store).

Due to type safety all non primitive type overlaps would result in conflicts that would result in build time errors. Unnecessary nesting makes state management & diffing more complex and imo therefore is no benefit, but if it gets this pr merged would change it :shrug:

Reply:
If its going to cause too much conflict in debugging down the road, then I see value in _not adding_ it in, but I don't have the experience yet debugging zustand.  I wasn't aware of the type safety checking that would happen if two methods with the same signature are part of the global store.  I'm a big fan of keeping things simple stupid and clean, but also like the idea of the scoping in the context of the primitives, and even though at this point the store may be simple and there aren't any overlaps, there are unknowns as to any issues that could arise if more things are added in to assist with this UI.  However, I supposed though if it becomes an issue down the line then we can always revisit it and key them and make the necessary updates across the board.  After our talk today and learning about how a store has access to other parts of inherited stores then I'm okay with the flattened idea, but again, I don't know what it's like debugging this state with and without the top level keys.